### ISSUE TITLE
[Suggestion] Chip RNG and Tweaks

### ISSUE THREAD

**Issue by Xerma** [tsunamayo/Starship-EVO#2852]
**Chip RNG Stats**

For starters I personally think the chip system is pretty neat but does pose some problems. I think there needs to be a lot more variety with chips. My example being that since each chip essentially has 3 positive stats that it changes there should be a "Balance Triangle" sort of thing going on. For example the Legendary Hull Upgrade Chip has these stats:
- Heat Dissipation: +78%
- Armor Point: +85%
- Integrity Threshold: +93%

These stats should first be brought down a lot to the point where they would form an equilateral triangle, like in this image:
![balance_triangle](https://user-images.githubusercontent.com/32974671/91667869-60b6a800-eacd-11ea-870a-e2cc7a22536d.png)

As an example then, that same chip that I mentioned above would instead be given BASE stats of:
- Heat Dissipation: +33%
- Armor Point: +33%
- Integrity Threshold: +33%

These would be the normal Legendary chip stats. But, each time a chip has the chance to drop, if it does drop, then that chip should be randomized like that of the balance triangle. So then the very best you could do for chips is to have one or two of those percentages increase while the third decreases. If both Heat Dissipation and Armor Point are at +50% then Integrity Threshold would be +0%. Now if just Heat Dissipation increases then the other two decrease so Heat Dissipation could be +100% while Armor Point and Integrity Threshold are at +0%.

**Further Chip Tweaks**

That all being said I personally also feel like chip stats should be based around what category or CAT the ship is. So each chip would have varying stats depending if it's going into a CAT 1 ship vs a CAT 5 ship. I feel this change would help keep what you wanted with giving smaller ships the ability to have a better chance vs larger ships without making larger ships just that much more terrifying and without taking away the ability for large ships to also have these bonuses.

**Chip Rerolling**

This was just a quick idea that came into mind where players would have this simple "crafting" system or reroll system where they could take say 3 chips they don't care for and consume those chips to reroll and produce one chip to try and get better stats instead of constantly grinding to get a chip that is the same rarity but has different stats on it. Those 3 chips would have to be of the same rarity and chip type in order to be rerolled. This idea relies on the first change proposed above.

**Comment by Whackjob** [comment]
Well, if we're going to nerf the chips, and they need it, that isn't a bad system.  The ship categories... eh.  I'm so far outside of those, I really don't know what to think.  Can we have a category 6 ship for capital ships, with the understanding that no ship yard will ever hold them?  I'll second the rerolling, though I'd be happy to just be able to delete or recycle the old ones.

**Comment by tsunamayo** [comment]
@Xerma yes for sure the system will require balancing. But I wont balance it yet in the void, I will wait for a proper gameplay first.
I dont like making it more complex with the category to be honest, that seems unnecessary.
I made the high tier stats like because I want to give advantage to smaller ship.
This way players will give their legendary chip to their smaller ship in their fleet. They have a smaller number of slots, so this helps. Whereas for larger ship with more slot, you can simply put several lower tiers chip for a similar effect.
For small vs big, well I am concerned about PvE. Except for bosses the NPCs wont have a lot of high tier chips!
Then for PvP, well your 20m ship is dead against a bigger 100m one anyway. If it was not the case I can already hear the complain of all the player who had the 100m ship saying the game is broken ;)

**Comment by tsunamayo** [comment]
Nice graph btw ;) cheers

**Comment by Xerma** [comment]
Oh yea I totally understand all that. The only reason I suggested the varying chip stats and reducing their stats as their category increases is because a normal ship has lets say 300,000 heat dissipation and that legendary chip would bring that up to about 534,000 heat dissipation. Whereas if a larger ship gets that same chip then their stats increase a ton. For example a ship I have that's still being built has roughly 3,000,000 heat dissipation (still well within survival limitations) so that chip would bring me to about 5,000,000 heat dissipation. So having reduced percentages for larger category ships would help keep the balance but still provide nice bonuses.

The category difference doesn't need to be dramatic but just noticeably different. 

If we were to only get one aspect of this suggestion I'd greatly prefer the whole balance triangle with varying chip stats. Instead of players gaining loads of the same chips they would have many various chips with different stats which could fit a wide variety of needs and desires when planning and building ships or other entities that have different purposes. I just feel that would keep players intrigued as to what chips they acquired from a battle rather than acquiring all of them and be like, oh yay, more chips that I already have.

And I didn't make that graph :P I just found it on google and used it because it showed exactly what I was meaning lol
